pkg/resource/endpoint/custom_update_api.go

```diff
@@ -0,0 +1,123 @@
+package endpoint
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/eventbridge"
+)
+
+// customUpdateEndpoint deals with the eventually consistent API of Endpoints in
+// a generic manner. It will always requeue a resource after modification to
+// sync its spec and status.
+//
+// note: there are several edge cases in this API which we don't cover e.g.,
+// going from a DISABLED replication state to ENABLED with an invalid ARN. The
+// controller currently has no logic to reconcile this against the eventual
+// consistent nature and inconsistent behavior of the API (API replication config will remain
+// DISABLED and state UPDATE_FAILED and does not reflect the desired state)
+func (rm *resourceManager) customUpdateEndpoint(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	if immutableFieldChanges := rm.getImmutableFieldChanges(delta); len(immutableFieldChanges) > 0 {
+		msg := fmt.Sprintf("Immutable Spec fields have been modified: %s", strings.Join(immutableFieldChanges, ","))
+		return nil, ackerr.NewTerminalError(fmt.Errorf(msg))
+	}
+
+	if err = validateEndpointSpec(delta, desired.ko.Spec); err != nil {
+		return nil, ackerr.NewTerminalError(err)
+	}
+
+	if endpointInMutatingState(latest) {
+		return latest, requeueWaitWhileUpdating
+	}
+
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	// we need to explicitly unset nil spec values
+	unsetRemovedSpecFields(delta, desired.ko.Spec, input)
+
+	_, err = rm.sdkapi.UpdateEndpointWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateEndpoint", err)
+	if err != nil {
+		return nil, err
+	}
+	return desired, ackrequeue.NeededAfter(nil, defaultRequeueDelay)
+}
+
+// newUpdateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+func (rm *resourceManager) newUpdateRequestPayload(
```

Comment:
If you use the standard generated update code path, this method would be autogenerated for you.